python/tank/platform/software_launcher.py

```diff
@@ -0,0 +1,738 @@
+        """
+        Allow the icon path to be set post-construction.
+
+        :param new_path: New path to icon
+        """
+        self._icon_path = new_path
+
+
+def _apply_version_to_setting(raw_string, version=None):
+    """
+    Replace any version tokens contained in the raw_string
+    with the appropriate version value from the app settings.
+    Replaces {version} and {v0}, {v1}, etc. tokens in raw_string
+    with their values. The {v} tokens are created by using groups
+    defined by () within the version string. For example, if the
+    version setting is "(9.0)v4(beta1)"
+        {version} = "9.0v4"
+        {v0} = "9.0"
+        {v1} = "beta1"
+
+    If version is None, we return the raw_string since there's
+    no replacement to do.
+
+    :param raw_string: the raw string potentially containing the
+                       version tokens (eg. {version}, {v0}, ...)
+                       we will be replacing. This string could
+                       represent a number of things including a
+                       path, an args string, etc.
+    :param version: version string to use for the token replacement.
+
+    :returns: string with version tokens replaced with their
+              appropriate values
+    """
+    # Verify there's something to replace.
+    if not raw_string or not version:
+        return raw_string
+
+    # split version string into tokens defined by ()s
+    version_tokens = re.findall(r"\(([^\)]+)\)", version)
+
+    # ensure we have a clean complete version string without ()s
+    clean_version = re.sub("[()]", "", version)
+
+    # do the substitution
+    ver_string = raw_string.replace("{version}", clean_version)
+    for i, token in enumerate(version_tokens):
+        ver_string = ver_string.replace("{v%d}" % i, token)
+    return ver_string
+
+
+class LaunchInformation(object):
+    """
+    Stores blueprints for how to launch a specific DCC.
+    """
+    def __init__(self, path=None, args=None, environ=None):
+        """
+        Constructor
+
+        :param path: Resolved path to DCC
+        :param args: Args to pass on the command line when launching the DCC
+        :param environ: Dict of environment variables : value that must be
+                        set to successfully launch the DCC.
+        """
+        # Initialize members
+        self._path = path
+        self._args = args or ""
+        self._environment = environ or {}
+
+    @property
+    def path(self):
+        """
+        The DCC path to launch
+
+        :returns: String path
+        """
+        return self._path
+
+    @property
+    def args(self):
+        """
+        Arguments to pass to the DCC on launch
+
+        :returns: String args
+        """
+        return self._args
+
+    @property
+    def environment(self):
+        """
+        Dictionary of environment variables to set before
+        launching DCC.
+
+        :returns: Dict {env_var: value}
+        """
+        return self._environment
```

Comment:
After the refactor, this file went from 728 lines to 406.  Is that still too big?  Should the SoftwareVersion and LaunchInformation classes be split out into their own files?